### PR TITLE
Adding functionality to create teams, and changing UI for dashboard (leaving and creating/joining teams)

### LINKED ISSUE #3313: In user profile (or dashboard), make it so users can create an organization
As discussed here (https://github.com/ProjectSidewalk/SidewalkWebpage/issues/2338#issuecomment-776769593), it would be nice if users could create their own organizations. Could they do this in their own dashboard/profile screen? This would take care of cases like https://github.com/ProjectSidewalk/SidewalkWebpage/issues/3312.

Easiest would just be to allow a person to join one and only one organization (I mention this because of https://github.com/ProjectSidewalk/SidewalkWebpage/issues/2338#issuecomment-777777882).

Similarly, it would be nice if users could see their own usernames and emails (and possibly change this information)

### PR DESCRIPTION
**Pull request by aslassi777**
Resolves #3313 

@misaugstad updated pr! will start working on #3054 throughout the week.

Added a new modal to help create teams. Added new backend to accommodate this request. Also added a few translations for the new dashboard across the website (including the modal form itself). Moved around UI per discussion with Michael. Also added the new escape HTML to the create team form, and did various renaming changes (most notably org to team).

##### Before/After screenshots 

Before:
Screenshot 2024-10-24 at 5 45 46 PM

After:

Screenshot 2024-10-24 at 5 47 35 PM

Screenshot 2024-10-16 at 4 35 06 PM

##### Things to check before submitting the PR 

- [x] I've written a descriptive PR title. 
- [x] I've added/updated comments for large or confusing blocks of code.
- [x] I've included before/after screenshots above.
- [x] I've asked for and included translations for any user facing text that was added or modified.
- [x] I've updated any logging. Clicks, keyboard presses, and other user interactions should be logged. If you're not sure how (or if you need to update the logging), ask Mikey. Then make sure the documentation on [this wiki page](https://github.com/ProjectSidewalk/SidewalkWebpage/wiki/Descriptions-of-Logged-Events) is up to date for the logs you added/updated.
- [x] I've tested on mobile (only needed for validation page).

### CODE REVIEW

File: app/models/user/OrganizationTable.scala
Comment (misaugstad) [inline app/models/user/OrganizationTable.scala:69]: Linking back to comment from previous review! Totally understand how it could get lost in the shuffle, given all the dev env issues!
https://github.com/ProjectSidewalk/SidewalkWebpage/pull/3696#discussion_r1807028368

File: app/views/leaderboard.scala.html
Comment (misaugstad) [inline app/views/leaderboard.scala.html:11]: `.headOption` is redundant here since it's no longer a list!

File: app/views/leaderboard.scala.html
Comment (misaugstad) [inline app/views/leaderboard.scala.html:244]: `.head` -> `.get`

File: app/views/userProfile.scala.html
Comment (misaugstad) [inline app/views/userProfile.scala.html:8]: You added this up here, but there are still a bunch of places where you could be using it and you aren't! I imagine that you could do the same thing with the output from `OrganizationTable.getOrganizationName()` as well!

And do another pass through to do the refactoring after switching from a `List[Int]` to `Option[Int]`! Some of the methods _work_, but are either redundant or aren't the preferred method:
* `head` -> `get`
* `nonEmpty` -> `isDefined`
* `headOption` is redundant

File: conf/messages.es
Comment (misaugstad) [inline conf/messages.es:412]: Really minor, but makes sure all the files end with newlines!

File: conf/messages.zh-TW
Comment (misaugstad) [inline conf/messages.zh-TW:68]: Looks like when merging changes from develop, you may have done more of a "keep my version" rather than a true "merge"! There are a couple lines changed here that shouldn't be. IIRC I changed lines only in the zh-TW file...

File: app/views/userProfile.scala.html
Comment (misaugstad) [inline app/views/userProfile.scala.html:35]: Let's switch to lowercase "strong"!

**Review by misaugstad — CHANGES_REQUESTED** [review]
Getting closer! I think that some stuff got lost in the shuffle with the dev environment issues, no worries! Aside from comments throughout the code, the only other change I'd make is to not make it looks like you can join a team that you're already in!
Screenshot from 2024-10-28 16-20-52

**Comment by aslassi777** [comment]
Sounds good @misaugstad! Going to make those final changes and should be good to go! Thanks for the review

File: app/views/userProfile.scala.html
Comment (aslassi777) [reply app/views/userProfile.scala.html:35]: sorry, this must have been a change lost in the dev process. fixed

**Comment by aslassi777** [comment]
@misaugstad should be good to go now! please let me know if anything else needs to be changed! :)

File: conf/routes
Comment (misaugstad) [inline conf/routes:155]: So minor, but it's just not quite in line with all the other routes :sweat_smile:

File: public/javascripts/Progress/src/Progress.js
Comment (misaugstad) [inline public/javascripts/Progress/src/Progress.js:57]: From [style guide](https://github.com/ProjectSidewalk/SidewalkWebpage/wiki/Style-guide): `if(...)){` -> `if (...) {`. One day I'll get the automated linter working!!

File: public/javascripts/Progress/src/Progress.js
Comment (misaugstad) [inline public/javascripts/Progress/src/Progress.js:85]: Wait I assume that this line does nothing, right?

File: public/javascripts/Progress/src/Progress.js
Comment (misaugstad) [inline public/javascripts/Progress/src/Progress.js:64]: Something isn't working quite right here.
* If I have no team and create a new one, it logs `Click_module=create_team=`. Missing the team id! Plus I think that it should  probably log both that you created it AND that you joined it!
* Joining a team doesn't log anything
* Switching team logs the leaving of a team correctly, but not the joining of the new team

But leaving a team is logged correctly!

Not sure how much of this was broken before you got here, but we've gotta fix it as we see it rather than adding more broken pieces on top!

**Review by misaugstad — CHANGES_REQUESTED** [review]
Oops, we ended up with a bunch of tiny changes still :sweat_smile: User-facing functionality is perfect though!

**Comment by aslassi777** [comment]
agreed @misaugstad! working on these changes right now

File: conf/messages.zh-TW
Comment (aslassi777) [reply conf/messages.zh-TW:338]: Oh i think I must have copy pasted the colon from google translator! ill fix!

File: public/javascripts/Progress/src/Progress.js
Comment (aslassi777) [reply public/javascripts/Progress/src/Progress.js:85]: yeah, that line was just moved downwards in the final its still there!

**Comment by aslassi777** [comment]
@misaugstad finished with all changes! Logging should work now for the different cases (creating teams (while already in a team/not in a team), leaving team, switching between teams, etc.

**Comment by aslassi777** [comment]
@misaugstad pulled in everything from develop! everything works on my end and my branch is up to date... feel free to take a look now! everything you requested should be fixed :)

File: public/javascripts/Progress/src/Progress.js
Comment (misaugstad) [inline public/javascripts/Progress/src/Progress.js:57]: The main issue with this is that you're declaring `endTeam` in two different places, which isn't great. This would be written more simply as a one-liner:
```
var endTeam = newTeam ? newTeam : parseId[3];
```

And this has the benefit of making it so that you're not declaring `endTeam` twice!

File: public/javascripts/Progress/src/Progress.js
Comment (misaugstad) [inline public/javascripts/Progress/src/Progress.js:68]: Since you removed the current team from the list of teams to join, it looks like we don't need to have the `endTeam != startTeam` check anymore!

File: public/javascripts/Progress/src/Progress.js
Comment (misaugstad) [inline public/javascripts/Progress/src/Progress.js:67]: I know you didn't write it this way, but let's most `window.location.reload();` to be after the logging. It really feels like the page load should be the last thing to be kicked off!

File: conf/messages.en
Comment (misaugstad) [inline conf/messages.en:330]: Didn't notice this last time around, but we actually have a translation for cancel, so you don't need to add one! It's worth doing a quick search through this file for a super common translation that we may have done before. It's near the top of the file, with just the key `cancel`!

File: conf/messages.de
Comment (misaugstad) [inline conf/messages.de:410]: It doesn't actually matter (because there are no issues with having two newlines at the ends of files), but you actually went from having no blank line at the ends of files to having two! Just wanted to point it out in case there is some issue with a setting in your IDE or something that you might want to look into? Again, not a big deal and I wouldn't bother to send you back to fix it lol

File: public/javascripts/Progress/src/Progress.js
Comment (misaugstad) [reply public/javascripts/Progress/src/Progress.js:81]: Not sure what you mean by that... I'm referring to the first line of this func, the line that's just:
```
escapeHTML
```

Seems like it's just the function name. It's not calling the function or anything as far as I can tell?

File: app/views/leaderboard.scala.html
Comment (misaugstad) [inline app/views/leaderboard.scala.html:11]: Since you made the change in `userProfile.scala.html` to replace the repeated uses of `UserOrgTable.getOrg()` and `OrganizationTable.getOrganizationName()` with a saved one at the top, could you do that for this file too?? Might as well improve page performance when we see small stuff like this!

**Review by misaugstad — CHANGES_REQUESTED** [review]
I swear we have to be almost there!! A final thing or two to iron out, and then I found a couple things I missed in my last two passes :upside_down_face: That'll get better over time, of course. As you get more detail-oriented, there are fewer things for me to comment on, so I'm less likely to miss things in any given pass! Onto my one note that isn't embedded in code:

The logging is a bit weird. Here are the logs that don't make much sense to me:
* When you just leave your team (without switching to another), it correctly logs `leaving_team=<team>` but it _also_ logs `joining_team=0`. It should log that you're joining a team if you aren't joining a team!
* On the flip side, if you aren't on a team and you join one then it's logging `leaving_team=0`. Again, if you aren't leaving a team, let's just log nothing instead of logging with a 0!

**Comment by aslassi777** [comment]
@misaugstad sounds good. To be honest, since I'm currently working on #3054 and trying to make some more forward progress, I'll come back to this one in a bit since these are just some simple changes! :)

**Comment by misaugstad** [comment]
@aslassi777 yes, definitely! It definitely makes sense to not have to pull you out of the flow on what you're working on!

File: conf/messages.de
Comment (aslassi777) [reply conf/messages.de:410]: Must have been a mistake! Checked the .de file and yes, there were 2 newlines :)

File: public/javascripts/Progress/src/Progress.js
Comment (aslassi777) [reply public/javascripts/Progress/src/Progress.js:81]: Yes absolutely right! That line should not be there (does nothing). Removed! Sorry didn't know what you were referencing at first.

**Comment by aslassi777** [comment]
Mikey made the final changes! Just waiting on final approval so I can push this branch into #3054 and get the PR process for that branch started!

**Review by misaugstad — APPROVED** [review]
Great, looking good!! I made some very minor edits whitespace edits, and merging now! Thanks for all the revisions!!

**Pull request merged**
